src/main/java/com/ldt/rankup/util/AccountingUtils.java

```diff
@@ -86,18 +86,26 @@ private String getURI()
 
     private Connection getConn()
     {
-        if (conn == null)
+        try
         {
-            String uri = getURI();
-            try
-            {
-                conn = getDataSource(uri).getConnection();
-            }
-            catch (SQLException e)
+            if (conn == null || conn.isClosed())
             {
-                getPlugin().getLogger().warn("Could not get Connection", e);
+                String uri = getURI();
+
+                try
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1141)